### PR TITLE
Editor: Unify the more menu

### PR DESCRIPTION
**Pull request by youknowriad**
Related #52632

## What?

This unifies the MoreMenu component between the post and site editors. There are still the following differences between the two menus:

 - The welcome guides are different (we need to unify these separately)
 - There's an "Export theme" link in the site editor. Maybe there's a better place in the site editor for this link (outside of the editor frame)
 - There's a "Manage patterns" link in the post editor. Maybe we should remove this link or include it as well in the site editor.
 - The PreferencesModal of the post editor include two extra preferences: PrePublish flag (will be unified soon) and "use theme styles" toggle that could be implemented in the site editor as well.
 - The post editor has a "full screen" toggle that doesn't really make sense in the post editor. (not sure what to do with this one ultimately) 

In order to keep these small differences in place while sharing the same code, I've used the existing ToolsMoreMenuGroup Slot and introduced a new private and similar ViewMoreMenuGroup. The order of the menu items is slightly different than trunk because the menu items that come through the slot are inserter after the "built-in" menu items.

## Testing Instructions

1- Check the MoreMenu of both post and site editors.

### CODE REVIEW

**Comment by jasmussen** [comment]
Good PR, seems close. Post editor before and after:

post before
post after

The main difference here is the sorting of the tools, primarily. It really is a bit of a kitchen-sink, that section, so they don't have to be resorted, but a better sorting may be:

* Copy all blocks
* Welcome Guide
* Keyboard shortcuts
* Manage patterns
* Help

Site editor before and after:

site before
site after

Same sorting item. Fine to omit the fullscreen toggle, it doesn't exist for the site editor.

The big difference here is that now you can load sidebar plugins in the site editor, such as Jetpack. And Jetpack doesn't really do anything in the site editor, and it can't be pinned like it can in the post editor. Could we/should we omit this?

**Comment by youknowriad** [comment]
For the moment, the sorting of "Welcome Guide" and "Manage Patterns" can't be modified because they are specific to each editor so they come last.

For Jepatck, I kind of disagree, I think it should load in the site editor. If it's useful for the "page editor" in the post editor, it should be useful in the site editor too.

**Comment by youknowriad** [comment]
Also I'm pretty sure the jetpack change didn't not happen in this PR.

**Comment by youknowriad** [comment]
How useful is the "manage patterns" link?

**Comment by jasmussen** [comment]
> For Jetpack, I kind of disagree, I think it should load in the site editor. If it's useful for the "page editor" in the post editor, it should be useful in the site editor too.

I should've clarified, I'm not necessarily against including it. It just seems like a biggish change, at least conceptually, and perhaps something worth extracting. Or maybe it's fine? The prominence is managed, after all.

> How useful is the "manage patterns" link?

It used to be the only way to get to these in the post editor. In the site editor the link is not very useful. Given there's now a dedicated "My patterns" section in the inserter, I wonder if we can move the link to this dropdown? Or just to that section in general?

my patterns in inserter

File: packages/editor/src/components/more-menu/index.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+								}
+								shortcut={ displayShortcut.access( 'h' ) }
+							>
+								{ __( 'Keyboard shortcuts' ) }
+							</MenuItem>
+							<CopyContentMenuItem />
+							<MenuItem
+								icon={ external }
+								href={ __(
+									'https://wordpress.org/documentation/article/wordpress-block-editor/'
```
Comment (ntsekouras) [inline packages/editor/src/components/more-menu/index.js:127]: This is another small difference too. In site editor the url was 'https://wordpress.org/documentation/article/site-editor/'

Comment (youknowriad) [reply packages/editor/src/components/more-menu/index.js:127]: I think where currently placed, the "block-editor" help page is better. If the help was shown outside (in the site editor layout or hub or something), the site editor help page would have been better.

**Comment by ntsekouras** [comment]
There is something wrong with `Show button text labels` preference in site editor. It seems broken. In post editor seems fine.

https://github.com/WordPress/gutenberg/assets/16275880/1aa370ae-17cb-41d6-9211-6255bb1030a9

**Comment by youknowriad** [comment]
@ntsekouras yep, small mistake, it should be fixed now

**Review by ntsekouras — APPROVED** [review]
>How useful is the "manage patterns" link?

I guess it's more useful in non block themes. 

>Also I'm pretty sure the jetpack change didn't not happen in this PR.

Yes, this seems to have happened in some other unification PRs. I agree that they should be shown in both editors, although the inability to pin items is an inconsistency we should address(not here though).

Code wise this looks good to me, thanks!

**Comment by youknowriad** [comment]
> although the inability to pin items is an inconsistency we should address(not here though).

Oh I didn't know bout this inconsistency, I'll look into it separately.

**Comment by jameskoster** [comment]
> How useful is the "manage patterns" link?

It seems a bit random at this point. Why no "Manage templates", "Manage pages", etc? I'd remove it.

> There's an "Export theme" link in the site editor. Maybe there's a better place in the site editor for this link (outside of the editor frame)

Do we still need this? It feels more like a dev tool that might be better suited to plugins ([create block theme](https://wordpress.org/plugins/create-block-theme/))?

Re sorting, the menu feels very long... it might be good to migrate to the new dropdown menu component in the future and place plugins/tools in flyouts.

**Comment by youknowriad** [comment]
> Do we still need this? It feels more like a dev tool that might be better suited to plugins ([create block theme](https://wordpress.org/plugins/create-block-theme/))?

For me create-block-theme should be core kind of (not all of it) but I believe core should be able to export good themes right away.

**Comment by jameskoster** [comment]
Yes I suppose that's the point. It feels like one small part of the theme creation process, so a bit random on its own, and in this location. But if we plan to bring more theme tools and flows to core then totally fine to leave it.

**Comment by youknowriad** [comment]
I agree the position is very random.

**Review by jameskoster — APPROVED** [review]
Everything seems to be working fine for me. I think we could live without the "Manage patterns" link, but we don't necessarily have to remove it here.

**Comment by youknowriad** [comment]
Yes, I've been thinking to land this as is (without too much impact) and consider each of these particular links on their own.

**Comment by jameskoster** [comment]
On the plugins point, I wonder if it would be good to allow them to define a scope in which they can be accessed. e.g. "Always", "Template editing", "Focused editing", "Post editing", "Zoomed out", etc. I agree the Jetpack tools are not very useful when you're editing a template / pattern.

**Comment by youknowriad** [comment]
@jameskoster That's already possible. I believe some plugins are already doing it because the post editor can already edit all post types. So it's just a matter of checking the active post type `wp.data.select( 'core/editor' ).getCurrentPostType()`

**Pull request merged**